api-definitions/platform/synthetic-data-generations.yaml

```diff
@@ -0,0 +1,104 @@
+openapi: '3.0.2'
+info:
+  title: Synthetic Data Generation
+  version: '0.1.0'
+
+paths:
+  /synthetic-data-generations:
+    post:
+      summary: Initialize a Synthetic Data Generation job
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SDGJobBody'
+      responses:
+        "200":
+          description: A successfully submitted job
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+  /synthetic-data-generations/{job_id}:
+    get:
+      summary: Retrieve the status of a Synthetic Data Generation job
+      responses:
+        "200":
+          description: The status for the job
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+
+    delete:
+      summary: Cancel a running Synthetic Data Generation job
+      responses:
+        "200":
+          description: The status for the job after cancellation
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+
+  /synthetic-data-generations/tasks:
+    get:
+      summary: List the currently supported SDG tasks
+      responses:
+        "200":
+          description: The set of currently supported SDG tasks
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/SDGTaskDefinition'
+
+components:
+  schemas:
+    SDGJobBody:
+      type: object
+      required: ['output_directory', 'tasks', 'seed_data']
+      properties:
+        output_directory:
+          description: Location to place the output in
+          $ref: '../common/file-pointer.yaml#/schemas/DirectoryPointer'
+
+        tasks:
+          description: Mapping from task name to task config. The config will be validated against the task's config schema.
```

Comment:
I see there's the potential for multiple "tasks" within one job: what do you visualize that being long term? Would it be something along the lines potentially of being able to have a "generate" task, a "mix" tasks to do random mixing of that data, and/or potentially a "filter task" to then filter some of the data and that should still all live within the context of a SDGJob?

Reply:
Good question. I took this from the CLI args to the library we're thinking of for the generic platform SDG implementation. I think for `InstructLab`, we'll likely only ever run a single `Task` (dataset + config + generation algorithm) per job.